src/stores/EndpointConfig.tsx

```diff
@@ -0,0 +1,383 @@
+import { getLiveSpecsByLiveSpecId, getSchema_Endpoint } from 'api/hydration';
+import { useEntityType } from 'context/EntityContext';
+import { useEntityWorkflow } from 'context/Workflow';
+import { EndpointConfigStoreNames } from 'context/Zustand';
+import { GlobalSearchParams } from 'hooks/searchParams/useGlobalSearchParams';
+import produce from 'immer';
+import { isEmpty, isEqual, map } from 'lodash';
+import {
+    createContext as createReactContext,
+    ReactNode,
+    useContext,
+} from 'react';
+import { createJSONFormDefaults } from 'services/ajv';
+import { ENTITY, EntityWithCreateWorkflow, JsonFormsData, Schema } from 'types';
+import useConstant from 'use-constant';
+import { devtoolsOptions } from 'utils/store-utils';
+import { createStore, StoreApi, useStore } from 'zustand';
+import { devtools, NamedSet } from 'zustand/middleware';
+
+export interface EndpointConfigState {
+    endpointConfig: JsonFormsData;
+    setEndpointConfig: (endpointConfig: JsonFormsData) => void;
+
+    endpointConfigErrorsExist: boolean;
+    endpointConfigErrors: { message: string | undefined }[];
+
+    endpointSchema: Schema;
+    setEndpointSchema: (val: EndpointConfigState['endpointSchema']) => void;
+
+    // Hydration
+    hydrated: boolean;
+    setHydrated: (value: boolean) => void;
+
+    hydrationErrorsExist: boolean;
+    setHydrationErrorsExist: (value: boolean) => void;
+
+    // Misc.
+    stateChanged: () => boolean;
+    resetState: () => void;
+}
+
+const fetchErrors = ({ errors }: JsonFormsData): JsonFormsData['errors'] => {
+    let response: JsonFormsData['errors'] = [];
+
+    if (errors && errors.length > 0) {
+        response = response.concat(errors);
+    }
+
+    return response;
+};
+
+const filterErrors = (list: JsonFormsData['errors']): (string | undefined)[] =>
+    map(list, 'message');
+
+const populateEndpointConfigErrors = (
+    endpointConfig: JsonFormsData,
+    state: EndpointConfigState
+): void => {
+    const endpointConfigErrors = filterErrors(fetchErrors(endpointConfig));
+
+    state.endpointConfigErrors = endpointConfigErrors.map((message) => ({
```

Comment:
The `SectionError` component requires errors to be of the following format: `{ message: string | undefined }[]`. During the type annotation audit for this quarter, I would like to narrow the type annotations of `SectionError` props.